### PR TITLE
[ETHEREUM-CONTRACTS/SUBGRAPH/JS-SDK] | Issue #1220 | Migrate `scripts` to `ops-scripts` and `dev-scripts`

### LINKED ISSUE #1220: [ETHEREUM-CONTRACTS] Behold, dev-scripts is here!
It is increasingly confusing for external developers to understand the difference between ops-script and dev-scripts, below is a summary and we will start to organize our scripts accordingly.

# Technical Notes

## OG Deployer

Purpose: This is for internal testing and production deployment.

### Artifacts

- deploy-test-environment.js ("test" here means for local testing of superfluid framework and public test networks of superfluid framework)
 - deploy-framework.js (for public networks)
 - deploy-test-token.js (for public networks, specifically test networks)
 - deploy-super-token.js (for public networks)

## New Deployer

Purpose: Recomended for testing with Superfluid Framework

### Artifacts

- SuperfluidFrameworkDeployer.sol
- deploy-test-framework.js ("test" here means for external usage of superfluid framework)

## About Splitting SuperfluidFrameworkDeployer

Use external library and contract factory patterns, e.g.

```
        SuperfluidCoreDeployer.createTestGovernance();
```

## Action

Options:

(a) Split scripts into ops-scripts and dev-scripts
(b) Retain scripts naming but put a README saying it is an alias of "ops-scripts", while creating new dev-scripts folder for developers building on superfluid framework.

@d10r

### PR DESCRIPTION
**Pull request by 0xdavinchee**
Issue Reference: https://github.com/superfluid-finance/protocol-monorepo/issues/1220

### Moving Scripts
- Moved almost all scripts to ops-scripts
- Moved deploy-test-environment to dev-scripts
- Updated workflows and bash scripts accordingly (scripts => ops-scripts)
- Updated documentation within `.js` scripts accordingly (scripts => ops-scripts)
- Update internal imports to ensure it doesn't break anything (subgraph)
- Update README.md in `ethereum-contracts` (scripts => ops-scripts)
- Update tsconfig's in `ethereum-contracts`
- Delete `index.js` in old `scripts`

cc @d10r

### CODE REVIEW

**Comment by 0xdavinchee** [comment]
add migration from ops-scripts => dev-scripts

File: packages/ethereum-contracts/package.json [hunk 1]
```diff
@@ -21,11 +21,11 @@
         "/build/abi.js",
         "artifacts/contracts/**/*.json",
         "!/**/*.dbg.json",
-        "scripts/**/*",
+        "dev-scripts/**/*",
```
Comment (hellwolf) [inline packages/ethereum-contracts/package.json:24]: let's leave "ops-scripts" there still.

also if we add compatibility "scripts" placeholders, add her too.

**Review by hellwolf — APPROVED** [review]
Generally GTG. But there is a chatter that we may want to put placeholder scripts to warn devs about the breaking changes.

File: packages/ethereum-contracts/dev-scripts/deploy-test-framework.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ const {ethers} = require("hardhat");
 const SlotsBitmapLibraryArtifact = require("@superfluid-finance/ethereum-contracts/artifacts/contracts/libs/SlotsBitmapLibrary.sol/SlotsBitmapLibrary.json");
 const SuperfluidFrameworkDeployerArtifact = require("@superfluid-finance/ethereum-contracts/artifacts/contracts/utils/SuperfluidFrameworkDeployer.sol/SuperfluidFrameworkDeployer.json");
 
-const ERC1820Registry = require("./artifacts/ERC1820Registry.json");
+const ERC1820Registry = require("../ops-scripts/artifacts/ERC1820Registry.json");
```
Comment (hellwolf) [inline packages/ethereum-contracts/dev-scripts/deploy-test-framework.js:6]: add comments: this script is hardhat only for now.

**Comment by hellwolf** [comment]
image

ready

**Pull request merged**